docking_preparation.py

```diff
@@ -0,0 +1,287 @@
+                continue
+            if len(pdb_id) == 4:
+                ligand3D = [f for f in ligand3d_sdf if pdb_id in str(f.stem)][0]
+            else:
+                with io.EntryReader(str(assayed_sdf)) as rdr:
+                    e = rdr[0]
+                    ligname = e.identifier.split('-')[0]
+                    ligand3D = [f for f in ligand3d_sdf if ligname in str(f)]
+                    if len(ligand3D) < 1:
+                        continue
+                    else:
+                        ligand3D = ligand3D[0]
+            with io.EntryReader(str(ligand3D)) as rdr:
+                for e in rdr:
+                    e.attributes['$File'] = str(ligand3D.resolve())
+                    e.attributes['STRUCID'] = pdb_id
+                    wri.write(e)
+
+
+def _get_pdb_ligname_df(target='pde-10'):
+    pdb_files = list((Path('') / Path(target) / Path('tmp_aligned_for_MOE')).glob('*.pdb'))
+    sdf_files = list((Path('') / Path(target) / Path('tmp_2d_sdf')).glob('*.sdf'))
+    ligand3d_sdf = list(
+        (Path('') / Path(target) / Path('tmp_aligned_3d_sdf')).glob('*.sdf'))
+
+    structure_quality = Path(f'{target}_structure_quality.csv')
+    high_quality_strucs = []
+    if structure_quality.is_file():
+        structure_quality = pd.read_csv(f'{target}_structure_quality.csv')
+        high_quality_strucs = structure_quality[(structure_quality['RESOLUTION'] <= 2.5) &
+                                                (structure_quality['DENSITYFLAGS'] <= 5) &
+                                                (structure_quality['LIGCLASHES'] == 0) &
+                                                (structure_quality['LIGSYMCONTACT'] == 0)]['STRUCID'].to_list()
+
+    structure_dicts = []
+
+    for sdf_file in sdf_files:
+        strucid = sdf_file.stem.split('_')[0]
+        with io.EntryReader(str(sdf_file)) as rdr:
+            full_lname = rdr[0].identifier
+            if full_lname.startswith('RO'):
+                lname = full_lname.split('-')[0]
+            else:
+                lname = full_lname
+
+        pdb_file = [f for f in pdb_files if f.stem.startswith(strucid)][0]
+        template_file = None
+        for f in ligand3d_sdf:
+            if f.stem.startswith('RO') and f.stem.startswith(lname):
+                template_file = f
+                break
+            elif not f.stem.startswith('RO') and f.stem.startswith(strucid):
+                template_file = f
+                break
+
+        structure_dicts.append({'ligand': lname, 'full_ligand_name': full_lname, 'pdb_file': str(pdb_file),
+                                   'template_file': template_file, 'strucid': strucid})
+
+    df = pd.DataFrame(structure_dicts)
+
+    if high_quality_strucs:
+        df['is_high_quality'] = df['strucid'].isin(high_quality_strucs)
+    else:
+        df['is_high_quality'] = True
+    df.to_csv('pdb_ligand.csv', index=False)
+
+
+def _sanitize_pdb_files(target='pde-10'):
+    from rdkit import Chem
```

Comment:
Imports should happen at the top of the file. If they are conditional, they can get wrapped in the import.